File: src/eventbus/js/eventBus.js
export class EventBus {
  constructor() {
    this.handlers = {};
    this.events = [];
  }

  register(eventName, handler) {
    if (!this.handlers[eventName]) this.handlers[eventName] = [];
    this.handlers[eventName].push(handler);
  }

  emit(eventName, payload = null) {
    const entry = {
      timestamp: Date.now(),
      event: eventName,
      payload
    };
    this.events.push(entry);

    (this.handlers[eventName] || []).forEach(fn => fn(payload));
    return entry;
  }

  listEvents() {
    return this.events;
  }
}

File: src/eventbus/python/event_bus.py
import time

class EventBus:
    """
    Lightweight, public-safe event bus for:
    - Orchestrator
    - SDK
    - Telemetry
    - Future ZK-light triggers
    - BHBR/BHT Web3 handshake (stub)
    """

    def __init__(self):
        self.handlers = {}
        self.events = []

    def register(self, event_name, handler):
        if event_name not in self.handlers:
            self.handlers[event_name] = []
        self.handlers[event_name].append(handler)
    
    def emit(self, event_name, payload=None):
        entry = {
            "timestamp": time.time(),
            "event": event_name,
            "payload": payload
        }
        self.events.append(entry)

        for handler in self.handlers.get(event_name, []):
            handler(payload)

        return entry

    def list_events(self):
        return self.events
